### PR TITLE
CAT-1395 Added sprites and look sorting

### PR DESCRIPTION
**Pull request by bernadettespieler**
Michael Fischl was a member in my XP team (SW 15) and he implemented this user story.
Long press at one object/look and you have further menu items: "Move Up" and "Move Down"

### CODE REVIEW

File: catroid/src/org/catrobat/catroid/ui/fragment/LookFragment.java [hunk 1]
```diff
@@ -780,6 +788,34 @@ protected void showDeleteDialog() {
 		deleteLookDialog.show(getFragmentManager(), DeleteLookDialog.DIALOG_FRAGMENT_TAG);
 	}
 
+	private void moveLookDataDown() {
+		if (selectedLookPosition < lookDataList.size() - 1) {
+			LookData down_element = lookDataList.get(selectedLookPosition + 1);
```
Comment (aried3r) [inline catroid/src/org/catrobat/catroid/ui/fragment/LookFragment.java]: Please use camelCase, not snake_case.

File: catroid/src/org/catrobat/catroid/ui/fragment/LookFragment.java [hunk 2]
```diff
@@ -780,6 +788,34 @@ protected void showDeleteDialog() {
...
+			lookDataList.set(selectedLookPosition + 1, selectedLookData);
+			lookDataList.set(selectedLookPosition, down_element);
+			adapter.notifyDataSetChanged();
+		} else {
+			LookData down_element = lookDataList.get(0);
```
Comment (aried3r) [inline catroid/src/org/catrobat/catroid/ui/fragment/LookFragment.java]: Same here.

File: catroid/src/org/catrobat/catroid/ui/fragment/LookFragment.java [hunk 3]
```diff
@@ -780,6 +788,34 @@ protected void showDeleteDialog() {
...
+			lookDataList.set(0, selectedLookData);
+			lookDataList.set(selectedLookPosition, down_element);
+			adapter.notifyDataSetChanged();
+		}
+	}
+
+	private void moveLookDataUp() {
+		if (selectedLookPosition > 0) {
+			LookData up_element = lookDataList.get(selectedLookPosition - 1);
```
Comment (aried3r) [inline catroid/src/org/catrobat/catroid/ui/fragment/LookFragment.java]: Same here.

File: catroid/src/org/catrobat/catroid/ui/fragment/LookFragment.java [hunk 4]
```diff
@@ -780,6 +788,34 @@ protected void showDeleteDialog() {
...
+			lookDataList.set(selectedLookPosition - 1, selectedLookData);
+			lookDataList.set(selectedLookPosition, up_element);
+			adapter.notifyDataSetChanged();
+		} else {
+			LookData up_element = lookDataList.get(lookDataList.size() - 1);
```
Comment (aried3r) [inline catroid/src/org/catrobat/catroid/ui/fragment/LookFragment.java]: And here.

File: catroid/src/org/catrobat/catroid/ui/fragment/SpritesListFragment.java [hunk 5]
```diff
@@ -361,6 +373,34 @@ public void startDeleteActionMode() {
 		}
 	}
 
+	private void moveSpriteDown() {
+		if (spritePosition < spriteList.size() - 1) {
+			Sprite down_element = spriteList.get(spritePosition + 1);
```
Comment (aried3r) [inline catroid/src/org/catrobat/catroid/ui/fragment/SpritesListFragment.java]: Here

File: catroid/src/org/catrobat/catroid/ui/fragment/SpritesListFragment.java [hunk 6]
```diff
@@ -361,6 +373,34 @@ public void startDeleteActionMode() {
...
+			spriteList.set(spritePosition + 1, spriteToEdit);
+			spriteList.set(spritePosition, down_element);
+			spriteAdapter.notifyDataSetChanged();
+		} else {
+			Sprite down_element = spriteList.get(1);
```
Comment (aried3r) [inline catroid/src/org/catrobat/catroid/ui/fragment/SpritesListFragment.java]: I'll not mark all occurrences, but you know what to do :)

File: catroid/src/org/catrobat/catroid/ui/fragment/LookFragment.java (hunk 1 above)
Comment (robertpainsi) [inline catroid/src/org/catrobat/catroid/ui/fragment/LookFragment.java]: Checkstyle also covers snake case
@bernadettespieler: ...so please take a look at the jenkins result.

**Comment by bernadettespieler** [comment]
https://jenkins.catrob.at/view/All-Categories/view/Catroid/job/Catroid-CustomEmulator-Single-ClassAndPackage-Test/387/

**Comment by robertpainsi** [comment]
Use http://developer.android.com/reference/java/util/Collections.html#swap(java.util.List<?>, int, int)

**Pull request closed without merging**